gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BomConflictCheckTask.java

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import netflix.nebula.dependency.recommender.provider.RecommendationProviderContainer;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.TaskAction;
+
+
+public class BomConflictCheckTask extends DefaultTask {
+
+    private final File propsFile;
+
+    @Inject
+    public BomConflictCheckTask(File propsFile) {
+        this.propsFile = propsFile;
+    }
+
+    @Input
+    public final Map<String, String> getMavenBomRecommendations() {
+        return getProject().getExtensions()
+                .getByType(RecommendationProviderContainer.class)
+                .getMavenBomProvider()
+                .getRecommendations();
+    }
+
+    @InputFile
+    public final File getPropsFile() {
+        return propsFile;
+    }
+
+    @TaskAction
+    public final void checkBomConflict() {
+        Map<String, String> recommendations = getMavenBomRecommendations();
+        List<Conflict> conflicts = new LinkedList<>();
+        Set<String> artifacts = BaselineVersions.getResolvedArtifacts(getProject());
+        Map<String, String> resolvedConflicts = new HashMap<>();
+        BaselineVersions.checkVersionsProp(getPropsFile(),
+                (propName, propVersion) -> {
+                    String regex = propName.replaceAll("\\*", ".*");
+                    artifacts.stream()
+                            .filter(artifactName -> artifactName.matches(regex))
+                            .forEach(artifactName -> resolvedConflicts.put(artifactName, propName));
+
+                    recommendations.entrySet()
+                            .stream()
+                            .filter(entry -> entry.getKey().matches(regex))
+                            .forEach(entry -> {
+                                conflicts.add(new Conflict(propName, propVersion, entry.getKey(), entry.getValue()));
+                                resolvedConflicts.remove(entry.getKey());
+                            });
+                    return null;
+                });
+
+        //Critical conflicts are versions.props line that only override bom recommendations with same version
+        //so it should avoid considering the case where a wildcard also pin an artifact not present in the bom
+        List<Conflict> critical = conflicts.stream()
+                .filter(c -> c.getBomVersion().equals(c.getPropVersion()))
+                .filter(c -> !resolvedConflicts.containsValue(c.getPropName()))
```

Comment:
would be best to avoid `containsValue` since you already have a hashmap

Reply:
Can you make conflicts indexed over the artifact as well?

Reply:
why is it a map then? You're just building a set

Reply:
I am not, I need to be able to remove values based on the key

Reply:
luckily no need to remove values anymore!